### PR TITLE
Add test update threshold recommendations

### PR DESCRIPTION
**Pull request by bharathappali**
This PR adds the tests for #1092 

Note: Please merge #1092 before merging this PR

### CODE REVIEW

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py [hunk 1]
```diff
@@ -1444,3 +1444,249 @@ def test_invalid_list_recommendations_notification_codes(cluster_type: str):
...
+                    assert "config" in short_term_recommendation["recommendation_engines"]["cost"]
+
+                if INFO_PERFORMANCE_RECOMMENDATIONS_AVAILABLE_CODE in short_term_recommendation["notifications"]:
+                    assert "variation" in short_term_recommendation["recommendation_engines"]["performance"]
+                    assert "config" in short_term_recommendation["recommendation_engines"]["performance"]
+
+                short_term_recommendation_cost_notifications = short_term_recommendation["recommendation_engines"]["cost"]["notifications"]
+                short_term_recommendation_perf_notifications = short_term_recommendation["recommendation_engines"]["performance"]["notifications"]
+
+                assert CPU_REQUEST_OPTIMISED_CODE in short_term_recommendation_cost_notifications
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1641]: Move this to a function as it is repeated

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py [hunk 2]
```diff
@@ -1444,3 +1444,249 @@ def test_invalid_list_recommendations_notification_codes(cluster_type: str):
...
+                if j > 1439: # 15 days
+                    long_term_recommendation = data_section[str(end_time)]["recommendation_terms"]["long_term"]
+
+                if INFO_COST_RECOMMENDATIONS_AVAILABLE_CODE in short_term_recommendation["notifications"]:
+                    assert "variation" in short_term_recommendation["recommendation_engines"]["cost"]
+                    assert "config" in short_term_recommendation["recommendation_engines"]["cost"]
+
+                if INFO_PERFORMANCE_RECOMMENDATIONS_AVAILABLE_CODE in short_term_recommendation["notifications"]:
+                    assert "variation" in short_term_recommendation["recommendation_engines"]["performance"]
+                    assert "config" in short_term_recommendation["recommendation_engines"]["performance"]
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1636]: We need to validate the cpu or memory values here for all the terms

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py [hunk 3]
```diff
@@ -1444,3 +1444,249 @@ def test_invalid_list_recommendations_notification_codes(cluster_type: str):
 
         response = delete_experiment(json_file)
         print("delete exp = ", response.status_code)
+
+@pytest.mark.sanity
+def test_list_recommendations_cpu_mem_optimised(cluster_type: str):
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1449]: Add test description here and also update the md file with the test cases added

Comment (bharathappali) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1449]: @chandrams I have added the test description and updated the readme.

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py (hunk 2 above)
Comment (bharathappali) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1660]: @chandrams added the value validation based on current config as both should be same

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py (hunk 1 above)
Comment (bharathappali) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py]: @chandrams Thanks for the review. I have created a new function in utils to check this. Please lemme know if there are any changes needed.

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py [hunk 4]
```diff
@@ -1630,10 +1630,50 @@ def test_list_recommendations_cpu_mem_optimised(cluster_type: str):
                 if INFO_COST_RECOMMENDATIONS_AVAILABLE_CODE in short_term_recommendation["notifications"]:
                     assert "variation" in short_term_recommendation["recommendation_engines"]["cost"]
                     assert "config" in short_term_recommendation["recommendation_engines"]["cost"]
+                    assert short_term_recommendation["recommendation_engines"]["cost"]["config"]["requests"]["cpu"]["amount"] == OPTIMISED_CPU
+                    assert short_term_recommendation["recommendation_engines"]["cost"]["config"]["limits"]["cpu"]["amount"] == OPTIMISED_CPU
+                    assert short_term_recommendation["recommendation_engines"]["cost"]["config"]["requests"]["memory"]["amount"] == OPTIMISED_MEMORY
+                    assert short_term_recommendation["recommendation_engines"]["cost"]["config"]["limits"]["memory"]["amount"] == OPTIMISED_MEMORY
+
+                    if j > 671:
+                        assert "variation" in medium_term_recommendation["recommendation_engines"]["cost"]
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1643]: Please move all this as well into a function and pass the term object as a parameter to it.

Comment (bharathappali) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py]: Thanks for the review @chandrams. I have made the changes accordingly.

**Review by chandrams — APPROVED** [review]
LGTM

**Pull request merged**